### PR TITLE
feat(ckms): allow override of url in command line 

### PR DESCRIPTION
**Pull request by Manuthor**

### CODE REVIEW

File: crate/pkcs11/provider/src/tests.rs [hunk 1]
```diff
@@ -47,13 +47,16 @@ fn initialize_backend() -> Result<CkmsBackend, Pkcs11Error> {
             .await
             .unwrap();
```
Comment (Adamk93) [inline crate/pkcs11/provider/src/tests.rs:48]: it might panic

File: crate/pkcs11/provider/src/tests.rs [hunk 2]
```diff
@@ -47,13 +47,16 @@ fn initialize_backend() -> Result<CkmsBackend, Pkcs11Error> {
...
 
-        let kms_client = ctx.owner_client_conf.initialize_kms_client().unwrap();
+        let kms_client = ctx
+            .owner_client_conf
+            .initialize_kms_client(None, None)
+            .unwrap();
         create_keys(&kms_client).await.unwrap();
         load_p12().await.unwrap();
```
Comment (Adamk93) [inline crate/pkcs11/provider/src/tests.rs:55]: it might panic also here.

File: crate/client/src/config.rs [hunk 3]
```diff
@@ -256,15 +256,26 @@ impl ClientConf {
...
+            kms_server_url,
             self.kms_access_token.as_deref(),
             self.ssl_client_pkcs12_path.as_deref(),
             self.ssl_client_pkcs12_password.as_deref(),
             self.kms_database_secret.as_deref(),
-            self.accept_invalid_certs,
+            accept_invalid_certs,
             if let Some(certificate) = &self.verified_cert {
                 Some(Certificate(
                     X509Certificate::from_pem(certificate.as_bytes())?.to_der()?,
```
Comment (Adamk93) [inline crate/client/src/config.rs:281]: if this returns error, the in the upper scope might panic, as the result is often unwrapped.

File: crate/test_server/benches/benches.rs [hunk 4]
```diff
@@ -48,7 +48,9 @@ fn bench_rsa_create_keypair(c: &mut Criterion) {
             .get_or_try_init(start_default_test_kms_server)
             .await
             .unwrap();
-        ctx.owner_client_conf.initialize_kms_client().unwrap()
+        ctx.owner_client_conf
+            .initialize_kms_client(None, None)
+            .unwrap()
```
Comment (Adamk93) [inline crate/test_server/benches/benches.rs:53]: panic risk.

File: crate/test_server/benches/benches.rs [hunk 5]
```diff
@@ -239,7 +241,10 @@ fn bench_rsa_encrypt(
             .get_or_try_init(start_default_test_kms_server)
             .await
             .unwrap();
-        let kms_rest_client = ctx.owner_client_conf.initialize_kms_client().unwrap();
+        let kms_rest_client = ctx
+            .owner_client_conf
+            .initialize_kms_client(None, None)
+            .unwrap();
```
Comment (Adamk93) [inline crate/test_server/benches/benches.rs:247]: panic risk.

File: crate/test_server/benches/benches.rs [hunk 6]
```diff
@@ -271,7 +276,10 @@ fn bench_rsa_decrypt(
...
```
Comment (Adamk93) [inline crate/test_server/benches/benches.rs:282]: panic risk.

File: crate/pkcs11/provider/src/tests.rs (hunk 1 above)
Comment (Adamk93) [reply crate/pkcs11/provider/src/tests.rs:48]: its true that are tests. so maybe panicking is not a big deal.

File: crate/client/src/config.rs (hunk 3 above)
Comment (Adamk93) [reply crate/client/src/config.rs:281]: It is true that so far I see the risk of panicking only in test and benches. Idk if it is safe to panic there.

**Review by Adamk93 — COMMENTED** [review]
I added some comments about the unwraps. They are used only in `tests.rs` and `benches.rs`. These lines might panic. If it is ok to panic than, the review is ok, if not, I think it has to be changed.

File: crate/pkcs11/provider/src/tests.rs (hunk 1 above)
Comment (Manuthor) [reply crate/pkcs11/provider/src/tests.rs:48]: as discussed together, `unwrap()` are allowed only in tests and benches.

File: crate/client/src/config.rs (hunk 3 above)
Comment (Manuthor) [reply crate/client/src/config.rs:281]: the `unwrap_or` does not panic. It returns the contained [Some](vscode-file://vscode-app/usr/share/code/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html) value or a provided default.

File: crate/test_server/benches/benches.rs (hunk 4 above)
Comment (Manuthor) [reply crate/test_server/benches/benches.rs:53]: thanks for notifying it. As discussed, unwrap() are allowed only in tests and benches.

File: crate/test_server/benches/benches.rs (hunk 6 above)
Comment (Manuthor) [reply crate/test_server/benches/benches.rs:282]: thanks for notifying it. As discussed, unwrap() are allowed only in tests and benches.

File: crate/test_server/benches/benches.rs (hunk 5 above)
Comment (Manuthor) [reply crate/test_server/benches/benches.rs:247]: thanks for notifying it. As discussed, unwrap() are allowed only in tests and benches.

**Pull request merged**